### PR TITLE
HMOS ZzHorizontalProgressbar Library

### PR DESCRIPTION
**Pull request by vidyaakbar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,163 +1,121 @@
 # ZzHorizontalProgressBar
```
Comment (samuel-abhishek) [inline README.md:1]: Add the build & quality badges

File: entry/src/main/resources/base/layout/ability_main.xml [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      <me.zhouzhuo.zzhoritontalprogressbar.ZzHorizontalProgressBar
+          ohos:height="60vp"
+          ohos:width="match_parent"
+          ohos:id="$+id:pb"
+          ohos:left_margin="20vp"
+          ohos:top_margin="20vp"
+          ohos:right_margin="20vp"
+          ohos:padding="20vp"
+          />
+
```
Comment (samuel-abhishek) [inline entry/src/main/resources/base/layout/ability_main.xml:25]: Remove the unnecessary whitelines. Applies to all the java & xml files. Please check

File: entry/src/ohosTest/java/me/zhouzhuo/zzhorizontalprogressbardemo/ExampleOhosTest.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
+package me.zhouzhuo.zzhorizontalprogressbardemo;
```
Comment (samuel-abhishek) [inline entry/src/ohosTest/java/me/zhouzhuo/zzhorizontalprogressbardemo/ExampleOhosTest.java:17]: Add the applibgroup copyright

File: zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java [hunk 4]
```diff
@@ -0,0 +1,1182 @@
...
+import java.lang.annotation.Target;
+
+
+/**
+ * 水平进度条*
+ * Created by 周卓 on 2016/9/22.
+ */
+
+public class ZzHorizontalProgressBar extends Component implements DrawTask {
+
```
Comment (samuel-abhishek) [inline zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java:27]: Please remove the unnecessary whitelines...Applies to all the xml and java files.

File: zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java [hunk 5]
```diff
@@ -0,0 +1,1182 @@
...
+    public ZzHorizontalProgressBar(Context context) {
+        this(context, null);
+        init(null);
+    }
+
+    /**
+     * Second Constructor.
+     *
+     * @param context   context as parameter.
+     *
```
Comment (samuel-abhishek) [inline zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java]: Line gap is not required between each param

File: zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java [hunk 6]
```diff
@@ -0,0 +1,1182 @@
...
+
+/**
+ * 水平进度条*
+ * Created by 周卓 on 2016/9/22.
+ */
+
+public class ZzHorizontalProgressBar extends Component implements DrawTask {
+
+
+    private final HiLogLabel hiLogLabel = new HiLogLabel(0, 0, "ZzHorizontalProgressBar");
```
Comment (samuel-abhishek) [inline zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java]: make this a static final variable and use the approprate naming convention.

File: zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java [hunk 7]
```diff
@@ -0,0 +1,1182 @@
...
+            canvas.drawCircle(mPadding
+                    + secondProgressHeight / 2.0f, mPadding
+                    + secondProgressHeight / 2.0f, secondProgressHeight / 2.0f, mSecondGradientPaint);
+        }
+
+    }
+
+    private void drawProgressCircleOpenGradientl(int width, float percent, int progressHeight) {
+        int progressWidth = width - mPadding * 2;
+        float dx;
```
Comment (samuel-abhishek) [inline zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java]: initialization and declaration ofthe variable can be in a single statement. (This applies to all the places wherever applicable, please check)

Comment (samuel-abhishek) [reply zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java]: Please make sure the distance between variable initialization & its first usage is not more than 5 lines. (CheckStyle issue)

File: zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java [hunk 8]
```diff
@@ -0,0 +1,1182 @@
...
+                drawProgressRoundRectMode(canvas);
+                drawBorderRoundRect(canvas);
+                break;
+            default:
+                HiLog.debug(hiLogLabel, "default switch case in OnDraw");
+        }
+    }
+
+    /**
+     * 绘制半圆形进度.
```
Comment (samuel-abhishek) [inline zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java]: Add the comments in English. Applies to all the methods.

**Review by samuel-abhishek — CHANGES_REQUESTED** [review]
Please check the review comments and do the required changes

File: entry/src/main/resources/base/layout/ability_main.xml (hunk 2 above)
Comment (vidyaakbar) [reply entry/src/main/resources/base/layout/ability_main.xml:25]: done

File: entry/src/ohosTest/java/me/zhouzhuo/zzhorizontalprogressbardemo/ExampleOhosTest.java (hunk 3 above)
Comment (vidyaakbar) [reply entry/src/ohosTest/java/me/zhouzhuo/zzhorizontalprogressbardemo/ExampleOhosTest.java:17]: done

File: zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java (hunk 4 above)
Comment (vidyaakbar) [reply zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java:27]: done

File: zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java (hunk 5 above)
Comment (vidyaakbar) [reply zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java]: done

File: zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java (hunk 8 above)
Comment (vidyaakbar) [reply zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java]: done

File: zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java (hunk 6 above)
Comment (vidyaakbar) [reply zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java]: done

File: zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java (hunk 7 above)
Comment (vidyaakbar) [reply zzhoritontalprogressbar/src/main/java/me/zhouzhuo/zzhoritontalprogressbar/ZzHorizontalProgressBar.java]: done

File: README.md (hunk 1 above)
Comment (vidyaakbar) [reply README.md:1]: we will resolve this after the project run on master branch

**Review by samuel-abhishek — APPROVED** [review]
LGTM

**Review by sharathks84 — APPROVED** [review]
Okay to merge. Review is done by Archana

**Review by Gleelakrishna — APPROVED** [review]
OK

**Pull request merged**